tests/Aspire.Hosting.Tests/KestrelConfigTests.cs

```diff
@@ -72,28 +85,71 @@ public void MultipleKestrelHttpEndpointsKeepTheirNames()
             );
     }
 
+    [Fact]
+    public async Task ExplicitEndpointsResultInKestrelOverridesAtRuntime()
+    {
+        var resource = CreateTestProjectResource<ProjectWithMultipleHttpKestrelEndpoints>(
+            operation: DistributedApplicationOperation.Run,
+            callback: builder =>
+            {
+                builder.WithHttpEndpoint(5017, name: "ExplicitProxiedHttp");
+                builder.WithHttpEndpoint(5018, name: "ExplicitNoProxyHttp", isProxied: false);
+            });
+
+        // Even though we're in Run mode, we need to get the env in Publish mode since ports don't get allocated
+        var config = await EnvironmentVariableEvaluator.GetEnvironmentVariablesAsync(resource, DistributedApplicationOperation.Publish);
```

Comment:
Is this testing the runtime behavior or publish time behavior? You can manually set allocated ports to force that code path to execute.

Reply:
The intent here is to test Publish, but I didn't know how to allocate the port. As it stands, it's kind of hybrid, in the sense that the ExecutionContext is in RunMode, when we get to `SetKestrelUrlOverrideEnvVariables`, but we're passing Publish here. I'll look at allocating ports to be in full run mode.

Reply:
I added code to allocate the ports, but I'm not sure this is correct. Is it normal for the env variables to still contain unevaluated expressions like `{projectName.bindings.FirstHttpEndpoint.targetPort}`? Clearly, by the time they get injected in the running process, everything must be evaluated.